### PR TITLE
Add `hack/update-template-oraclelinux.sh`

### PR DESCRIPTION
**Pull request by norio-nomura**
```console
$ hack/update-template-oraclelinux.sh
update-template-oraclelinux.sh: Update the Oracle Linux image location in the specified templates

Usage:
  update-template-oraclelinux.sh [--version-major <major version>] <template.yaml>...

Description:
  This script updates the Oracle Linux image location in the specified templates.
  Image location basename format:

        OL<major version>U<minor version>_<arch>-kvm[-cloud]-b<build number>.qcow2

  Published Oracle Linux image information is fetched from the following URLs:

    OL8:
          x86_64: https://yum.oracle.com/templates/OracleLinux/ol8-template.json
          aarch64: https://yum.oracle.com/templates/OracleLinux/ol8_aarch64-cloud-template.json
          
        OL9:
          x86_64: https://yum.oracle.com/templates/OracleLinux/ol9-template.json
          aarch64: https://yum.oracle.com/templates/OracleLinux/ol9_aarch64-cloud-template.json

  The downloaded files will be cached in the Lima cache directory.

Examples:
  Update the Oracle Linux image location in templates/**.yaml:
  $ update-template-oraclelinux.sh templates/**.yaml

  Update the Oracle Linux image location to major version 9 in ~/.lima/oraclelinux/lima.yaml:
  $ update-template-oraclelinux.sh --version-major 9 ~/.lima/oraclelinux/lima.yaml
  $ limactl factory-reset oraclelinux

Flags:
  --version-major <major version>  Use the specified Oracle Linux <major version>.
                                   The major version must be 7+ for x86_64 or 8+ for aarch64.
  -h, --help                       Print this help message
```
This PR aims to add Oracle Linux support as per https://github.com/lima-vm/lima/issues/1347.
This change will likely conflict with #2811. After one is merged, the other will need to be rebased.

### CODE REVIEW

File: hack/update-template-oraclelinux.sh [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+Examples:
+  Update the Oracle Linux image location in templates/**.yaml:
+  $ $(basename "${BASH_SOURCE[0]}") templates/**.yaml
+
+  Update the Oracle Linux image location to version 9 in ~/.lima/oraclelinux/lima.yaml:
+  $ $(basename "${BASH_SOURCE[0]}") --version 9 ~/.lima/oraclelinux/lima.yaml
+  $ limactl factory-reset oraclelinux
+
+Flags:
+  --version <version>  Use the specified Oracle Linux <version>.
```
Comment (AkihiroSuda) [inline hack/update-template-oraclelinux.sh:46]: Shouldn't this be `--version-major`?
For consistency with alma and rocky

Comment (norio-nomura) [reply hack/update-template-oraclelinux.sh:46]: As far as I’ve found, Oracle Linux does not have major or minor versions, but instead uses descriptions like “version 8 update 10,” so I’ve aligned with that format.

Comment (AkihiroSuda) [reply hack/update-template-oraclelinux.sh:46]: Their "version" number equates to the major version of RHEL/Alma/Rocky, and their "update" number equates to the minor version.

Comment (AkihiroSuda) [reply hack/update-template-oraclelinux.sh:46]: They also call it "8.10": https://docs.oracle.com/en/operating-systems/oracle-linux/8/relnotes8.10/

> Release Notes for Oracle Linux 8.10

Comment (norio-nomura) [reply hack/update-template-oraclelinux.sh]: reworded `version` to `major version`, `release` to `minor version`

**Review by AkihiroSuda — APPROVED** [review]
Thanks

File: templates/oraclelinux-8.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # This template requires Lima v0.9.0 or later.
-# Oracle image licence: https://www.oracle.com/downloads/licenses/oracle-linux-license.html
+# Oracle image license: https://www.oracle.com/downloads/licenses/oracle-linux-license.html
```
[the remark this answers is not in the document]
Comment (norio-nomura) [reply templates/oraclelinux-8.yaml:2]: I didn't know. Thanks.

**Pull request merged**